packages/read-model/src/Product/Listed/ListedProductViewFactory.php

```diff
@@ -144,12 +145,20 @@ public function createFromProduct(Product $product, ?ImageView $imageView, Produ
      */
     public function createFromArray(array $productArray, ?ImageView $imageView, ProductActionView $productActionView, PricingGroup $pricingGroup): ListedProductView
     {
+        $productPrice = $this->priceFactory->createProductPriceFromArrayByPricingGroup(
+            $productArray['prices'],
+            $pricingGroup
+        );
+        if ($productPrice === null) {
+            throw new NoProductPriceForPricingGroupException($productArray['id'], $pricingGroup->getId());
```

Comment:
I'm thinking about throwing this exception directly in the factory. Because from a create method I expect creating an object and not return null. What do you think?

Reply:
I agree but we cannot do it in minor version as discussed.